services/protondb/protondb.service.js

```diff
@@ -0,0 +1,71 @@
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { BaseJsonService } from '../index.js'
+
+const documentation = `
+  <div>
+    <p>
+      You can obtain a user's ID once a <a href="https://www.protondb.com/help/site-questions#why-are-my-reports-not-showing-up-on-game-pages">content refresh</a>
+      has occurred after they've filed their first report. This can take a few days.
+    </p>
+    <p>
+      Once the user has a published report, their profile becomes publicly accessible by clicking on the left-side of their reports.
+      Profile URLs are formatted like <code>/users/:id</code>, take the ID from here.
+    </p>
+  </div>
+  `
+
+/**
+ * Validates that the schema response is what we're expecting.
+ */
+const schema = Joi.object({
+  reports: Joi.array().required(),
+}).required()
+
+/**
+ * This badge displays the total number of reports someone has submitted
+ * to ProtonDB.
+ */
+export default class ProtonDbPoints extends BaseJsonService {
+  static category = 'other'
+  static route = {
+    base: 'protondb/reports',
+    pattern: ':id',
+  }
+
+  static examples = [
+    {
+      title: 'ProtonDB reports',
+      namedParams: { id: '938767784' },
+      staticPreview: this.render({ reports: 12 }),
+      documentation,
+    },
+  ]
+
+  static defaultBadgeData = { label: 'reports', color: 'info' }
+
+  static render({ reports }) {
+    return { message: metric(reports) }
+  }
+
+  async fetch({ id }) {
+    return this._requestJson({
+      schema,
+      url: `https://www.protondb.com/data/users/by_id/${id}.json`,
```

Comment:
Is there any documentation for this API endpoint we're calling here?

Reply:
OK thanks. The reason we say

> Badges should not obtain data from undocumented or reverse-engineered API endpoints.

in [our guidelines](https://github.com/badges/shields/blob/master/CONTRIBUTING.md#badge-guidelines)

is because we have had some situations where a contributor worked out that a project had an API by inspecting the frontend code for their website or whatever, added a badge that calls it, and then we ended up relying on an API that wasn't really intended for public use. This then either annoyed the maintainers of the company/project behind it or we had no reasonable expectation of stability/compatibility over time (because devs maintaining the the API were making the assumption the only thing consuming it was their own frontend code) and stuff kept breaking.

That's pretty much the situation we've got here so I am going to close this PR. I would happily revisit this in future if there is a reliable public source for the data. It looks like this service meets all our other criteria.

Sorry.

Reply:
Ahh, I see! Sorry about that then!

I implemented the freeCodeCamp badge similarly to this one then. 😨 
Maybe we should review and remove that one from here too then?

Reply:
We know there are some that already exist - it is not something the maintainers were super strict on in the early days of the project, but it is something we think about a lot more now.

Prevention is better than cure but for stuff that already exists, we:
- accept the past has happened
- prioritise backwards-compatibility
- remove badges if there is a problem

so I wouldn't worry about badges that already exist.